### PR TITLE
Added coverage for CLI settings

### PR DESCRIPTION
**Pull request by ntkathole**
Test results 

```
pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_hostname_prefix_without_value
=============================================== test session starts ========================================
collected 9 items

tests/foreman/cli/test_setting.py .

========================================== 1 passed =========================================================
```

```
pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_hostname_default_prefix
=============================================== test session starts =========================================
collected 9 items

tests/foreman/cli/test_setting.py .

========================================= 1 passed ===========================================
```

```
pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_login_page_footer_text
============================================ test session starts ===============================================

collected 9 items


tests/foreman/cli/test_setting.py .

========================================= 1 passed =========================================
```

```

pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_login_page_footer_text_without_value
================================================= test session starts ========================================
collected 9 items


tests/foreman/cli/test_setting.py .

=============================================== 1 passed ========================================================
```

### CODE REVIEW

File: tests/foreman/cli/test_setting.py [hunk 1]
```diff
@@ -36,29 +40,32 @@ def test_negative_update_hostname_with_empty_fact(self):
...
         """Update the Hostname_prefix settings without any string(empty values)
 
         :id: a84c28ea-6821-4c31-b4ab-8662c22c9135
 
         :expectedresults: Hostname_prefix should be set without any text
-
-        :caseautomation: notautomated
         """
+        setting = Settings.set({'name': "discovery_prefix", 'value': ""})
+        self.assertIn("", setting[0])
```
Comment (abalakh) [inline tests/foreman/cli/test_setting.py]: This specific assertion doesn't really verify anything, "emptiness" is present in the string for sure :)
Anyway, pls avoid verifying messages. Prefer verifying actual value instead. Should be smth like this:
```python
discovery_prefix = Settings.list({'search': 'name=discovery_prefix'})[0]
self.assertEqual('', discovery_prefix['value'])

File: tests/foreman/cli/test_setting.py [hunk 2]
```diff
@@ -36,29 +40,32 @@ def test_negative_update_hostname_with_empty_fact(self):
...
         :expectedresults: Default set prefix should be updated with new value
-
-        :caseautomation: notautomated
         """
+        for hostname_prefix_value in valid_hostname_prefix():
+            with self.subTest(hostname_prefix_value):
+                setting = Settings.set({
+                                 'name': "discovery_prefix",
+                                 'value': hostname_prefix_value}
+                )
```
Comment (abalakh) [inline tests/foreman/cli/test_setting.py]: Wrong indentation and closing bracket placement.
```python
                setting = Settings.set({
                    'name': "discovery_prefix",
                    'value': hostname_prefix_value,
                })
```

File: tests/foreman/cli/test_setting.py (hunk 1 above)
Comment (abalakh) [reply tests/foreman/cli/test_setting.py]: Comment applicable to all the tests

File: tests/foreman/cli/test_setting.py [hunk 3]
```diff
@@ -100,11 +106,14 @@ def test_positive_update_login_page_footer_text(self):
...
 
         :expectedresults: Parameter is updated successfully
-
-        :caseautomation: notautomated
         """
+        for login_text_value in valid_data_list():
+            with self.subTest(login_text_value):
+                setting = Settings.set({'name': "login_text",
+                                        'value': login_text_value}
+                                       )
```
Comment (abalakh) [inline tests/foreman/cli/test_setting.py]: Same here

**Comment by ntkathole** [comment]
@abalakh and @ldjebran please revisit....updated Test results

```
pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_login_page_footer_text
========================================================== test session starts ==========================================================

collected 17 items


tests/foreman/cli/test_setting.py .

======================================================= 1 passed ==================================================
```

```

pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_login_page_footer_text_without_value
========================================================== test session starts ==========================================================

collected 17 items


tests/foreman/cli/test_setting.py .

======================================================= 1 passed in ==================================================
```


```
 pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_hostname_default_prefix
========================================================== test session starts ==========================================================

collected 17 items


tests/foreman/cli/test_setting.py .

======================================================= 1 passed  ======================================
```

```
pytest tests/foreman/cli/test_setting.py::SettingTestCase::test_positive_update_hostname_prefix_without_value
========================================================== test session starts ==========================================================

collected 17 items


tests/foreman/cli/test_setting.py .

======================================================= 1 passed ============================================
```

File: tests/foreman/cli/test_setting.py [hunk 4]
```diff
@@ -36,29 +40,37 @@ def test_negative_update_hostname_with_empty_fact(self):
...
 
         :id: a6e46e53-6273-406a-8009-f184d9551d66
 
         :expectedresults: Default set prefix should be updated with new value
-
-        :caseautomation: notautomated
         """
+        hostname_prefix_value = gen_string('alpha')
+        Settings.set({
+                         'name': "discovery_prefix",
```
Comment (abalakh) [inline tests/foreman/cli/test_setting.py]: Still wrong indentation. Should be only 4 spaces more than 'Settings', not 12:
```python
Settings.set({
    'name': "discovery_prefix",
    'value': hostname_prefix_value
})
```

**Pull request merged**